docs/source/user_guide/simple_pixie_c_lib.md

```diff
@@ -0,0 +1,490 @@
+  store float %6, float* %2, align 4, !tbaa !14
+  ret void
+}
+
+attributes #0 = { mustprogress nofree norecurse nosync nounwind ssp uwtable willreturn "frame-pointer"="non-leaf" "min-legal-vector-width"="0" "no-trapping-math"="true" "stack-protector-buffer-size"="8" }
+
+!llvm.module.flags = !{!0, !1, !2, !3, !4, !5, !6, !7, !8}
+!llvm.ident = !{!9}
+
+!0 = !{i32 1, !"wchar_size", i32 4}
+!1 = !{i32 1, !"branch-target-enforcement", i32 0}
+!2 = !{i32 1, !"sign-return-address", i32 0}
+!3 = !{i32 1, !"sign-return-address-all", i32 0}
+!4 = !{i32 1, !"sign-return-address-with-bkey", i32 0}
+!5 = !{i32 7, !"PIC Level", i32 2}
+!6 = !{i32 1, !"Code Model", i32 1}
+!7 = !{i32 7, !"uwtable", i32 1}
+!8 = !{i32 7, !"frame-pointer", i32 1}
+!9 = !{!"clang version 14.0.6"}
+!10 = !{!11, !11, i64 0}
+!11 = !{!"double", !12, i64 0}
+!12 = !{!"omnipotent char", !13, i64 0}
+!13 = !{!"Simple C/C++ TBAA"}
+!14 = !{!15, !15, i64 0}
+!15 = !{!"float", !12, i64 0}
+```
+
++++
+
+We can make a separate PIXIE library that references the `add_f64` function and have it optimized with the embedded bitcode.
+
++++
+
+Here's a C program that references `add_f64` in a loop:
+
+```{code-cell} ipython3
+%%writefile simple_add_loop.c
+
+/* external reference */
+void add_f64(double *x, double *y, double *out);
+
+void loop_add_f64(double *x, double *y, double *out, int size) {
+    for (int i=0; i<size; ++i)
+        add_f64(x, y, out);
+}
+```
+
+Compile it into a PIXIE library.
+
+```{code-cell} ipython3
+src = "simple_add_loop.c"
+tus = [
+    TranslationUnit.from_c_source(src),
+]
+export_config = ExportConfiguration()
+# Note that both C symbols are stored into the same Python name 
+# such that it is building an overloaded function (like C++).
+export_config.add_symbol(python_name='loop_add',
+                         symbol_name='loop_add_f64',
+                         signature='void(double*, double*, double*)',)
+compiler = PIXIECompiler(library_name='my_c_loop',
+                         translation_units=tus,
+                         export_configuration=export_config,
+                         **get_default_configuration(),
+                         python_cext=True,     # True to make a Python C-extension 
+                         output_dir='.')
+compiler.compile()
+```
+
+We will again get the bitcode from the new PIXIE library.
+
+```{code-cell} ipython3
+import my_c_loop
+mod_loop = llvm.parse_bitcode(my_c_loop.__PIXIE__['bitcode'])
+print(mod_loop)
+```
+
+Output:
+
+```llvm
+source_filename = "simple_add_loop.c"
+target datalayout = "e-m:o-i64:64-i128:128-n32:64-S128"
+target triple = "arm64-apple-macosx14.0.0"
+
+; Function Attrs: nounwind ssp uwtable
+define void @loop_add_f64(double* noundef %0, double* noundef %1, double* noundef %2, i32 noundef %3) local_unnamed_addr #0 {
+  %5 = icmp eq double* %2, %0
+  %6 = icmp eq double* %2, %1
+  %7 = or i1 %5, %6
+  %8 = icmp sgt i32 %3, 0
+  %9 = and i1 %7, %8
+  br i1 %9, label %10, label %14
```

Comment:
NOTE: At some point this needs updating to reflect the removal of the aliasing guard https://github.com/numba/pixie/pull/30#discussion_r1664471745. However, it can remain for now as it doesn't fundamentally alter the concepts being discussed and demonstrated.